storage/remote/queue_manager.go

```diff
@@ -1020,7 +1020,7 @@ func (t *QueueManager) shouldReshard(desiredShards int) bool {
 	}
 	// We shouldn't reshard if Prometheus hasn't been able to send to the
 	// remote endpoint successfully within some period of time.
-	minSendTimestamp := time.Now().Add(-2 * time.Duration(t.cfg.BatchSendDeadline)).Unix()
+	minSendTimestamp := time.Now().Add(-2*time.Duration(t.cfg.BatchSendDeadline) + shardUpdateDuration).Unix()
```

Comment:
this could potentially even just be `time.Now().Add(-1*shardUpdateDuration)`, if we haven't been able to successfully send since the last time we checked if we _should_ reshard then just skip

Reply:
@kushalShukla-web just checking in, I think we should make the change I've suggested above :point_up: and then merge your pr :+1:

Reply:
sure @cstyan  !